### PR TITLE
[JIRA] fix optional props for creating issues

### PR DESCRIPTION
**Pull request by andrewjschuang**
Applies implemented fixes by @mattgauf in #2911 on the refactored Jira action component.

Co-authored-by: mattgauf mattgauf@gmail.com
Resolves #2911.

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4010"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/jira/actions/create-issue/create-issue.mjs [hunk 1]
```diff
@@ -199,8 +218,8 @@ export default {
     let properties;
     try {
       properties = JSON.parse(this.properties);
-    } catch ( err ) {
-      //pass
+    } catch (err) {
```
Comment (vellames) [inline components/jira/actions/create-issue/create-issue.mjs:221]: If we fail to parse we just ignore it? There are no side effects of doing this?

**Review by vellames — CHANGES_REQUESTED** [review]
Hey, just left one comment

File: components/jira/actions/create-issue/create-issue.mjs (hunk 1 above)
Comment (andrewjschuang) [reply components/jira/actions/create-issue/create-issue.mjs:221]: Yeah, I guess. Added error logging

**Comment by mattgauf** [comment]
@andrewjschuang I like the refactoring, this is much better than my original patchwork solution.

**Comment by andrewjschuang** [comment]
> @andrewjschuang I like the refactoring, this is much better than my original patchwork solution.

Thanks, but it was not me haha it was done by @feyzullah. I just added your requirements on top of the refactoring

**Comment by mattgauf** [comment]
> > @andrewjschuang I like the refactoring, this is much better than my original patchwork solution.
> 
> Thanks, but it was not me haha it was done by @feyzullah. I just added your requirements on top of the refactoring

Oh gotcha! Still, thanks for following through on this. I'm glad I could contribute!

**Comment by DilanAthukorala** [comment]
Verification completed and the issue has been resolved. This is ready for release.

**Comment by andrewjschuang** [comment]
Merging on behalf of @mattgauf.
Thanks again for the contribution!

**Pull request merged**